src/Controller/Api/TwoFactorAuthController.php

```diff
@@ -73,102 +62,34 @@
      * )
      *
      * @Route(
-     *     path="/api/2fa",
-     *     name="api_security_2fa_enable",
+     *     path="/api/2fa/activate",
+     *     name="api_security_2fa_activate",
      *     methods={"POST"}
      * )
      *
      * @param Request                $request
      * @param EntityManagerInterface $manager
-     * @param ErrorNormalizer        $errorNormalizer
      *
-     * @return FormInterface|JsonResponse
+     * @return FormInterface|Response
      */
-    public function enable(Request $request, EntityManagerInterface $manager, ErrorNormalizer $errorNormalizer)
+    public function activateTwoFactor(Request $request, EntityManagerInterface $manager)
     {
         /** @var User $user */
         $user = $this->getUser();
         if ($user->getGoogleAuthenticatorSecret()) {
-            return new JsonResponse(['errors' => ['Two-factor authentication code already enabled']], Response::HTTP_BAD_REQUEST);
+            return Response::create(null, Response::HTTP_CREATED);
         }
 
         $form = $this->createForm(TwoFactoryAuthEnableType::class, $user);
         $form->submit($request->request->all());
-        if ($form->isValid()) {
-            $manager->persist($user);
-            $manager->flush();
-
-            return new JsonResponse();
+        if (!$form->isValid()) {
+            return $form;
         }
 
-        return new JsonResponse($errorNormalizer->normalize($form), Response::HTTP_BAD_REQUEST);
-    }
-
-    /**
-     * Disable 2FA on your account.
-     *
-     * @SWG\Tag(name="Security")
-     *
-     * @SWG\Parameter(
-     *     name="body",
-     *     in="body",
-     *     required=true,
-     *     @SWG\Schema(
-     *         type="object",
-     *         properties={
-     *             @SWG\Property(
-     *                 property="_auth_code",
-     *                 type="integer"
-     *             )
-     *         }
-     *     )
-     * )
-     * @SWG\Response(
-     *     response=200,
-     *     description="Success disable 2FA"
-     * )
-     * @SWG\Response(
-     *     response=401,
-     *     description="Unauthorized"
-     * )
-     *
-     * @SWG\Response(
-     *     response=400,
-     *     description="Returns errors",
-     *     @SWG\Schema(
-     *         type="object",
-     *         @SWG\Property(
-     *             type="string",
-     *             property="errors"
-     *         )
-     *     )
-     * )
-     *
-     * @Route(
-     *     path="/api/2fa",
-     *     name="api_security_2fa_disable",
-     *     methods={"DELETE"}
-     * )
-     *
-     * @param Request                      $request
-     * @param EntityManagerInterface       $manager
-     * @param GoogleAuthenticatorInterface $twoFactor
-     *
-     * @return FormInterface|JsonResponse
-     */
-    public function disable(Request $request, EntityManagerInterface $manager, GoogleAuthenticatorInterface $twoFactor)
```

Comment:
How i can disable 2FA?

Reply:
Enable always for now.

Reply:
I think we should have optional this functionality
@beshkenadze confirm, please